src/webgpu/web_platform/copyToTexture/canvas.spec.ts

```diff
@@ -393,16 +535,121 @@ g.test('copy_contents_from_gl_context_canvas')
     // Construct expected value for different dst color format
     const dstBytesPerPixel = kTextureFormatInfo[dstColorFormat].bytesPerBlock;
     const format: RegularTextureFormat = formatForExpectedPixels(dstColorFormat);
-    const expectedPixels = t.getExpectedPixels({
-      context: canvasContext,
+    const sourcePixels = t.getSourceCanvasGLContent(canvasContext, width, height);
+    const expectedPixels = t.getExpectedPixels(
+      sourcePixels,
       width,
       height,
       format,
-      contextType: 'gl',
+      srcPremultiplied,
+      dstPremultiplied
+    );
+
+    t.doTestAndCheckResult(
+      { source: canvas, origin: { x: 0, y: 0 } },
+      {
+        texture: dst,
+        origin: { x: 0, y: 0 },
+        colorSpace: 'srgb',
+        premultipliedAlpha: dstPremultiplied,
+      },
+      { width: canvas.width, height: canvas.height, depthOrArrayLayers: 1 },
+      dstBytesPerPixel,
+      expectedPixels,
+      isFp16Format(dstColorFormat)
+    );
+  });
+
+g.test('copy_contents_from_gpu_context_canvas')
+  .desc(
+    `
+  Test HTMLCanvasElement and OffscreenCanvas with webgpu context
+  can be copied to WebGPU texture correctly.
+
+  It creates HTMLCanvasElement/OffscreenCanvas with 'webgpu'.
+  Use stencil + clear to render red rect for top-left, green rect
+  for top-right, blue rect for bottom-left and white for bottom-right.
+  And do premultiply alpha in advance if the webgl/webgl2 context is created
+  with premultipliedAlpha : true.
```

Comment:
Update the comment on lines 570-573 (test actually uses writeTexture).

Also the WebGL version of this description should say "scissor" not "stencil"